platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/impl/ConfigurationAdminImpl.java

```diff
@@ -578,9 +578,15 @@ public ObjectClassDefinition getObjectClassDefinition(String pid) {
             boolean ocdRequired) throws InvalidSyntaxException {
         List<Service> serviceList = new ArrayList<>();
 
+        // service.pid will contain service.factoryPid, so it is best to filter with service.pid
+        String newFilter = null;
```

Comment:
❓ I'm still a little confused on why this part is necessary. Could you clarify what we're trying to achieve here?